### ISSUE TITLE
Validator gets stuck in endless loop

### ISSUE THREAD

**Issue by Silverlan** [KhronosGroup/glTF-Validator#141]
I've tried validating the attached glTF file, but the validator just gets stuck at "Validating..." endlessly and never seems to progress. The file does not contain any meshes, only a skeleton and an animation.
I've only tried the web-version of the validator (https://github.khronos.org/glTF-Validator/).

[soldier.zip](https://github.com/KhronosGroup/glTF-Validator/files/4446149/soldier.zip)

**Comment by lexaknyazev** [comment]
Confirmed.

**Comment by javagl** [comment]
Just in case that it *might* be helpful: The same appeared for a model that is simpler than the attached "soldier" model:

<sup>(Note that the following model does not have a buffer URI, it was only extracted from a GLB, but this seems to be unrelated to the issue)</sup>

```
{
  "scene":0,
  "scenes":[
    {
      "name":"Scene",
      "nodes":[
        0
      ]
    }
  ],
  "nodes":[
    {
      "mesh":0,
      "name":"Mesh"
    }
  ],
  "meshes":[
    {
      "name":"Mesh",
      "primitives":[
        {
          "indices":0,
          "attributes":{
            "POSITION":1,
            "TEXCOORD_0":2,
            "COLOR_0":3,
            "NORMAL":4
          }
        }
      ]
    }
  ],
  "accessors":[
    {
      "bufferView":0,
      "componentType":5125,
      "count":12,
      "max":[
        4
      ],
      "min":[
        0
      ],
      "type":"SCALAR"
    },
    {
      "bufferView":1,
      "componentType":5126,
      "count":5,
      "max":[
        1.0,
        1.0,
        2.0
      ],
      "min":[
        -1.0,
        -1.0,
        -0.0
      ],
      "type":"VEC3"
    },
    {
      "bufferView":2,
      "componentType":5126,
      "count":5,
      "max":[
        1.0,
        1.0
      ],
      "min":[
        0.0,
        0.0
      ],
      "type":"VEC2"
    },
    {
      "bufferView":3,
      "componentType":5126,
      "count":5,
      "max":[
        1.0,
        1.0,
        1.0
      ],
      "min":[
        0.0,
        0.0,
        0.0
      ],
      "type":"VEC3"
    },
    {
      "bufferView":4,
      "componentType":5126,
      "count":5,
      "max":[
        0.5773503184318543,
        0.5773503184318543,
        1.0
      ],
      "min":[
        -0.5773503184318543,
        -0.5773503184318543,
        0.5773503184318543
      ],
      "type":"VEC3"
    }
  ],
  "bufferViews":[
    {
      "buffer":0,
      "byteOffset":0,
      "byteLength":48,
      "target":34963
    },
    {
      "buffer":0,
      "byteOffset":48,
      "byteLength":220,
      "byteStride":44,
      "target":34962
    },
    {
      "buffer":0,
      "byteOffset":60,
      "byteLength":208,
      "byteStride":44,
      "target":34962
    },
    {
      "buffer":0,
      "byteOffset":68,
      "byteLength":200,
      "byteStride":44,
      "target":34962
    },
    {
      "buffer":0,
      "byteOffset":80,
      "byteLength":188,
      "byteStride":44,
      "target":34962
    }
  ],
  "buffers":[
    {
      "byteLength":268
    }
  ]
}
```

(This is taken from a "GLB with an unusual data layout", as described in https://github.com/KhronosGroup/glTF-Asset-Generator/issues/563#issuecomment-651467159 - maybe this helps to narrow down the search space for the error)

**Comment by lexaknyazev** [comment]
The original issue caused by a loop in a node graph and has been fixed internally some time ago, I just need to sync the public GitHub (KTX2 ratification took all the resources last few months).

The snippet above is certainly unrelated to that and does not cause an endless loop (it's still invalid as it lacks an `asset` object).

**Comment by javagl** [comment]
Sorry, originally it included the `asset`, but I removed it because I overlooked that it was *required*  while I tried to remove elements from the file in order to figure out what caused the error. 

It's hard (at least for me) to figure out the exact *reason* of the error. I only added it here because the *observed behavior* was the same, namely that it says "Validating..." and does not print any error messages or so. If it's really unrelated, I can post this as a separate issue.
